### DISCUSSION TITLE
UndefinedTable by indexing

### DISCUSSION THREAD

**Discussion by Isyaam** [osm-search/Nominatim#3089] [Q&A]
Hello, I'm having a problem after importing my database.

I created a .env file with the following:

NOMINATIM_DATABASE_DSN="pgsql:hostaddr=127.0.0.1;port=5432;dbname=nominatim;user=nominatim"
NOMINATIM_FLATNODE_FILE="./node.cache"

Next, I started the import with the following command:

nominatim import --no-updates --threads 8 --osm-file europe-latest.osm.pbf 2>&1 | tee setup.log

I didn't encounter any import errors. 
But after the import, when I try running 'nominatim index', I get the following error:

Starting rank 12 (using batch size 1)

psycopg2.errors.UndefinedTable: relation "search_name_168" does not exist
LINE 1: DELETE from search_name_168 WHERE place_id = in_place_id
                                      ^
QUERY: DELETE from search_name_168 WHERE place_id = in_place_id
CONTEXT: PL/pgSQL function deletesearchname(integer,bigint) line 826 at SQL statement
PL/pgSQL function placex_update() line 44 at assignment

What did I wrong?

**Comment by lonvia** [comment]
You have made an non-updatable import (using `--no-updates`). You cannot run any commands that change the database afterwards including `nominatim index`. Why do you want to run this command?

**Reply by Isyaam** [reply]
When I test the database with

`nominatim admin --check-database`

I encounter a failure with the index, and I receive this index command.

Is this command necessary for all operations in Nominatim?

**Reply by mtmail** [reply]
You're right, the `--check-database` should not make the recommendation to run `nominatim --index` in this case. I'll get that fixed.

**Reply by mtmail** [reply]
> Is this command necessary for all operations in Nominatim?

It's not. Having a couple, even a couple of hundred places not indexed is fine. All places got indexed at least once during the initial import. The `--check-database` is too strict.
